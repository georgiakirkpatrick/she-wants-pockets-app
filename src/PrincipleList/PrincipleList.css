.PrincipleList {
    background-color: gray;
    list-style-type: none;
}