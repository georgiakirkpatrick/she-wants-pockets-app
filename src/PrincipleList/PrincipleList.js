import React from 'react';
import Principle from '../Principle/Principle';
import './PrincipleList.css';
import ReadMoreLink from '../ReadMoreLink/ReadMoreLink';

function PrincipleList(props) {
    const generatePrinciples = props.principles.map(principle => (
        <Principle
            key={principle.id}
            id={principle.id}
            // icon={principle.icon} 
            title={principle.title} 
            description={principle.description}
        />
    ))
    
    return (
        <div>
             <ul className='PrincipleList'>
                {generatePrinciples}
            </ul>

            <ReadMoreLink 
                link='/principles' 
                text='Learn more about our selection criteria'
            />
        </div>
       
    )
}

PrincipleList.defaultProps = {
    principles: []
}

export default PrincipleList;