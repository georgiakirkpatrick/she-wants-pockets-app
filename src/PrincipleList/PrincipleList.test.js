import React from 'react';
import ReactDOM from 'react-dom';
import PrincipleList from './PrincipleList';
import renderer from 'react-test-renderer';

const principles = [
    {
        id: 1821,
        title: 'Good design', 
        description: 'We value versatile styles that bring out our best selves in the office and at happy hour.  That hold up on our bike commutes and school pick-ups.  Our ideal clothing is bra-friendly, easy care, and has pockets.'
    },
    {
        id: 1822,
        title: 'Ethical',
        description: 'We report which brands have the most transparent supply chains, and which brands prioritize worker safety, living wages, and freedom of association (the right to join a union).'
    },
    {
        id: 1823,
        title: 'Sustainable',
        description: 'We feature clothing made with natural and cellulose-based fibers, with exceptions for a few categories like outerwear and swimwear.  While we include thoughtfully-sourced wool products, we exclude leather and fur.'
    }
]

describe ('PrincipleList', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<PrincipleList />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<PrincipleList principles={principles} />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})