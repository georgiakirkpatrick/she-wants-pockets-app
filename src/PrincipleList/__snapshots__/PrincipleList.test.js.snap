// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrincipleList renders the UI as expected 1`] = `
<div>
  <ul
    className="PrincipleList"
  >
    <li
      className="Principle"
    >
      <h3>
        Good design
      </h3>
      <p>
        We value versatile styles that bring out our best selves in the office and at happy hour.  That hold up on our bike commutes and school pick-ups.  Our ideal clothing is bra-friendly, easy care, and has pockets.
      </p>
    </li>
    <li
      className="Principle"
    >
      <h3>
        Ethical
      </h3>
      <p>
        We report which brands have the most transparent supply chains, and which brands prioritize worker safety, living wages, and freedom of association (the right to join a union).
      </p>
    </li>
    <li
      className="Principle"
    >
      <h3>
        Sustainable
      </h3>
      <p>
        We feature clothing made with natural and cellulose-based fibers, with exceptions for a few categories like outerwear and swimwear.  While we include thoughtfully-sourced wool products, we exclude leather and fur.
      </p>
    </li>
  </ul>
  <a
    href="/principles"
  >
    <p
      className="ReadMoreLink"
    >
      <span>
        Learn more about our selection criteria
         
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-right fa-w-14 "
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </p>
  </a>
</div>
`;
